src/Test/StateMachine/Types.hs

```diff
@@ -117,3 +125,14 @@ toPair :: (a, a) -> Pair a
 toPair (x, y) = Pair x y
 
 type ParallelCommands = ParallelCommandsF Pair
+
+type NParallelCommands = ParallelCommandsF []
```

Comment:
Hmm, I wonder if using length indexed lists, i.e. `Vec :: Nat -> Type`, instead of lists here would make things easier? Would make it possible to write `toPairUnsafe` in a safe way, but could also make other things more complicated...

Reply:
Indeed, I think writing shrinkers may be complicated. Also, I only use the unsafe functions in tests.